File: app/src/main/java/com/github/gfranks/collapsible/calendar/sample/MainActivity.java
package com.github.gfranks.collapsible.calendar.sample;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import com.github.gfranks.collapsible.calendar.CollapsibleCalendarView;

import org.joda.time.LocalDate;
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends Activity implements CollapsibleCalendarView.Listener<Event> {

    private CollapsibleCalendarView mCalendarView;
    private ListView mListView;
    private EventListAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mCalendarView = findViewById(R.id.calendar);
        mListView = findViewById(R.id.calendar_event_list);

        mCalendarView.setListener(this);
        mCalendarView.addEvents(getEvents());
        mCalendarView.setMinDate(new LocalDate(2010, 1, 1));
        mCalendarView.setMaxDate(new LocalDate(2020, 1, 1));
        Log.d("TAG", "min date" + mCalendarView.getMinDate().toString());
        Log.d("TAG", "max date" + mCalendarView.getMaxDate().toString());
    }

    private List<Event> getEvents() {
        List<Event> events = new ArrayList<>();
        for (int i=-20; i<80; i++) {
            events.add(new Event("Event " + (i+1), System.currentTimeMillis() + (21600000 * i)));
        }
        return events;
    }

    @Override
    public void onDateSelected(LocalDate date, List<Event> events) {
        if (mAdapter == null || mListView.getAdapter() == null) {
            mAdapter = new EventListAdapter(this, events);
            mListView.setAdapter(mAdapter);
        } else {
            mAdapter.setEvents(events);
        }
    }

    @Override
    public void onMonthChanged(LocalDate date) {
    }

    @Override
    public void onHeaderClick() {
        mCalendarView.toggle();
    }

    private class EventListAdapter extends ArrayAdapter<String> {

        public final DateTimeFormatter mTimeFormat = DateTimeFormat.forPattern(" h:mm a");
        private List<Event> mEvents;

        public EventListAdapter(Context context, List<Event> events) {
            super(context, android.R.layout.simple_list_item_1);
            mEvents = events;
        }

        public void setEvents(List<Event> events) {
            mEvents = events;
            notifyDataSetChanged();
        }

        @Override
        public int getCount() {
            return mEvents.size();
        }

        @Override
        public String getItem(int position) {
            Event event = mEvents.get(position);
            return mTimeFormat.print(event.getListCellTime()) + " - " + event.getTitle();
        }
    }
}

File: app/src/main/java/com/github/gfranks/collapsible/calendar/sample/MainActivity.kt
package com.github.gfranks.collapsible.calendar.sample

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.widget.ArrayAdapter
import com.github.gfranks.collapsible.calendar.CollapsibleCalendarView
import com.github.gfranks.collapsible.calendar.model.CollapsibleCalendarEvent
import kotlinx.android.synthetic.main.activity_main.*
import org.joda.time.LocalDate
import org.joda.time.format.DateTimeFormat
import org.joda.time.format.DateTimeFormatter
import java.util.*

class MainActivity : Activity(), CollapsibleCalendarView.ICollapsibleCalendarListener {

    private var mAdapter: EventListAdapter? = null

    private val events = ArrayList<Event>()

    init {
        for (i in -20..79) {
            events.add(Event("Event " + (i + 1), System.currentTimeMillis() + 21600000 * i))
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        calendar.apply {
            setListener(this@MainActivity)
            addEvents(events)
            minDate = LocalDate(2010, 1, 1)
            maxDate = LocalDate(2020, 1, 1)
        }
    }

    override fun onDateSelected(date: LocalDate, events: List<CollapsibleCalendarEvent>) {
        if (mAdapter == null || calendar_event_list.adapter == null) {
            mAdapter = EventListAdapter(this, ArrayList(events))
            calendar_event_list.adapter = mAdapter
        } else {
            mAdapter?.setEvents(events)
        }
    }

    override fun onMonthChanged(date: LocalDate?) {}

    override fun onHeaderClick() {
        calendar.toggle()
    }

    private inner class EventListAdapter(context: Context, private val mEvents: ArrayList<CollapsibleCalendarEvent>) : ArrayAdapter<String>(context, android.R.layout.simple_list_item_1) {

        val mTimeFormat: DateTimeFormatter? = DateTimeFormat.forPattern(" h:mm a")

        fun setEvents(events: List<CollapsibleCalendarEvent>) {
            mEvents.clear()
            mEvents.addAll(events)
            notifyDataSetChanged()
        }

        override fun getCount(): Int = mEvents.size

        override fun getItem(position: Int): String? {
            val event = mEvents[position] as Event
            return mTimeFormat?.print(event.listCellTime) + " - " + event.title
        }
    }
}
